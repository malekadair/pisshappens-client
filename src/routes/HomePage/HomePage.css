.HomePage {
  font-size: 2em;
  margin-top: 54px;
}

.ready {
  font-size: 0.8em;
  color: #003f5a;
  text-align: center;
  margin: 0 auto;
  letter-spacing: 2px;
}

.guess-button {
  color: #003f5a !important;
  text-transform: uppercase;
  font-weight: 600;
  letter-spacing: 0.1em;
  padding: 5px 15px;
  margin-top: 3rem;
  border: 3px solid #003f5a !important;
  border-radius: 6px;
  display: inline-block;
  transition: all 0.3s ease 0s;
  position: relative;
  top: 100%;
  left: 50%;
  -ms-transform: translate(-50%, -50%);
  transform: translate(-50%, -50%);
}

.guess-button:hover {
  background-color: #ebd9c8;
  border-radius: 15px;
  border-color: #003f5a !important;
  transition: all 0.3s ease 0s;
  cursor: pointer;
}

/* ----------- iPhone X ----------- */

@media only screen and (min-device-width: 375px) and (max-device-width: 812px) and (-webkit-min-device-pixel-ratio: 3) {
  .ready {
    font-size: 0.5em;
    letter-spacing: 1px;
  }
  .guess-button {
    font-size: 0.5em;
  }
}
