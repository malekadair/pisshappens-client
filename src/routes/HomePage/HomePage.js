import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import Typical from 'react-typical'
import Welcome from '../../components/Welcome/Welcome';
import WinnersList from '../../components/WinnersList/WinnersList'
import Nav from '../../components/Nav/Nav';
import TokenService from '../../services/token-service'
import WinnersApiService from '../../services/winners-api-service'
import WinnersContext from '../../contexts/winnersContext'

import './HomePage.css'
import Footer from '../../components/Footer/Footer';

class HomePage extends Component {
	static contextType = WinnersContext

	componentDidMount() {
		this.context.clearError();
		WinnersApiService.getAllWinners()
			.then(this.context.setWinners)
			.catch(this.context.setError)
	}

	render() {
		return (
			<div>
				<header>
					<Nav />
				</header>
				<main>
					<div className='HomePage'>

						{TokenService.hasAuthToken()
							? <WinnersList />
							: <Welcome />}

						<Typical
							steps={["Ready to submit your own guess?", 5000, "", 2000]}
							loop={Infinity}
							wrapper="h2"
							className='ready'
						/>
						<div>
							<Link to='/guess' className='guess-button'>Guess!</Link>
						</div>
					</div>
				</main>
				<Footer />
			</div >
		)
	}
}

export default HomePage