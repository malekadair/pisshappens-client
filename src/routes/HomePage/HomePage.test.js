import React from "react";
import toJson from "enzyme-to-json";
import HomePage from "./HomePage";

describe(`HomePage component`, () => {
	const props = {
		id: "a",
		name: "test-class-name"
	};

	it("renders a HomePage by default", () => {
		const wrapper = <HomePage />;
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("renders the HomePage given props", () => {
		const wrapper = <HomePage {...props} />;
		expect(toJson(wrapper)).toMatchSnapshot();
	});
});