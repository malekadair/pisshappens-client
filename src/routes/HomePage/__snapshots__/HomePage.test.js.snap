// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component renders a HomePage by default 1`] = `undefined`;

exports[`HomePage component renders the HomePage given props 1`] = `undefined`;
